web/client-api/src/main/java/io/deephaven/web/client/api/JsTable.java

```diff
@@ -666,21 +667,32 @@ private Promise<JsTotalsTable> fetchTotals(Object config, JsProvider<ClientTable
             JsArray<String> dropColumns = directive.getDropColumns();
             requestMessage.setSourceId(target.getHandle().makeTableReference());
             requestMessage.setResultId(newState.getHandle().makeTicket());
-            if (updateViewExprs != null && updateViewExprs.length != 0) {
+            if (updateViewExprs.length != 0) {
                 SelectOrUpdateRequest columnExpr = new SelectOrUpdateRequest();
                 columnExpr.setResultId(requestMessage.getResultId());
                 requestMessage.setResultId();
                 columnExpr.setColumnSpecsList(updateViewExprs);
-                TableReference prev = new TableReference();
-                prev.setBatchOffset(0);
-                columnExpr.setSourceId(prev);
+                columnExpr.setSourceId(new TableReference());
+                columnExpr.getSourceId().setBatchOffset(0);
```

Comment:
Was this causing a problem previous? Just seems odd to rewrite it this way, would expect to set the batchOffset before calling `setSourceId`.

Reply:
No, this is just a line shorter this way. The JS api for pb isn't immutable, so we can do it this way, and while it does offer a builder pattern (setter methods return `this`), the TS that is generated only offers plain setters, so they can't be chained (and the java is generated from the TS).